int16 mode
----
bool result